mica-search/src/main/java/org/obiba/mica/file/search/rest/AbstractFileSearchResource.java

```diff
@@ -0,0 +1,59 @@
+package org.obiba.mica.file.search.rest;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.obiba.mica.web.model.Mica;
+
+import javax.inject.Inject;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.obiba.mica.core.service.PublishedDocumentService;
+import org.obiba.mica.file.AttachmentState;
+import org.obiba.mica.web.model.Dtos;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+
+import static org.obiba.mica.file.FileUtils.normalizePath;
+
+public abstract class AbstractFileSearchResource {
+  protected static final String DEFAULT_SORT = "name";
+  protected static final String MAX_SIZE = "999";
+
+  protected abstract PublishedDocumentService getDocumentService();
+
+  protected String getQueryString(String path, String query, boolean recursively) {
+    String pathPart = String
+      .format("path:%s%s", !Strings.isNullOrEmpty(path) ? QueryParser.escape(normalizePath(path)) : "",
+        recursively ? "\\/*" : "");
+    String queryString = Joiner.on(" AND ").join(
+      Stream.of(pathPart, !Strings.isNullOrEmpty(query) ? String.format("(%s)", QueryParser.escape(query)) : null)
+        .filter(s -> s != null).iterator());
+
+    return queryString;
+  }
+
+  @Inject
+  private Dtos dtos;
+
+  @GET
+  @Path("/{path:.*}")
+  public List<Mica.FileDto> searchFile(@PathParam("path") String path, @QueryParam("query") String query,
+    @QueryParam("recursively") @DefaultValue("false") boolean recursively, @QueryParam("from") @DefaultValue("0") int from,
+    @QueryParam("limit") @DefaultValue(MAX_SIZE) int limit, @QueryParam("sort") @DefaultValue(DEFAULT_SORT) String sort,
+    @QueryParam("order") @DefaultValue("desc") String order) {
+    String queryString = getQueryString(path, query, recursively);
+    PublishedDocumentService.Documents<AttachmentState> states = getDocumentService()
+      .find(from, limit, sort, order, null, queryString);
+
+    return states.getList().stream().map(state -> dtos.asFileDto(state, state.isPublished()))
```

Comment:
asFileDto() 2nd param is not the publication status of the AttachmentState, it is the context of the execution: draft file system or published file system.
In the draft file system you return more info in the dto. In the published file system you return less info and results must be published files.